### PR TITLE
LPS-162412 [PRM] Open Claim form in MDFRequest Detail

### PR DESCRIPTION
**Pull request by DiogoNo**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+const currentPath = Liferay.currentURL.split('/');
+const mdfRequestedId = +currentPath[currentPath.length - 1];
```
Comment (thaynam) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js:14]: mdfRequestdId

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const SITE_URL = Liferay.ThemeDisplay.getLayoutRelativeURL()
+	.split('/')
+	.slice(0, 3)
+	.join('/');
+
+const claimButton = fragmentElement.querySelector('.newClaim');
+
+claimButton.onclick = () => {
+	Liferay.Util.navigate(
+		`${SITE_URL}/marketing/mdf-claim/new/#/mdfrequest/${mdfRequestedId}`
```
Comment (thaynam) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js:24]: mdfRequestId

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js (hunk 1 above)
Comment (gabsprates) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js:14]: Thinking about the result of our build process, and browser support, is it possible using [`Array.prototype.at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)?

Comment (gabsprates) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/new-claim-button/index.js:14]: Hmm, I just noticed this is a fragment. Does fragments code goes through some (e.g.: babel) build processes?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/useGetMDFRequestById.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ import MDFRequestDTO from '../../../../interfaces/dto/mdfRequestDTO';
 import {LiferayAPIs} from '../../common/enums/apis';
 import liferayFetcher from '../../common/utils/fetcher';
 
-export default function useGetMDFRequestById(id: number | undefined) {
+export default function useGetMDFRequestById(id: number | string | undefined) {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/useGetMDFRequestById.ts:19]:
```suggestion
export default function useGetMDFRequestById(id: number | undefined) {
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by jackelyneaquino** [comment]
jackelyneaquino

**Comment by jackelyneaquino** [comment]
jackelyneaquino

**Pull request closed without merging**